---
title: "Ontology Analysis"
output:
  html_document:
    df_print: paged
  pdf_document: default
---
```{r "setup", include=FALSE}
#Set the WD here
knitr::opts_chunk$set(fig.width=16, fig.height=12) 
library(clusterProfiler)
library(DOSE)
options(connectionObserver = NULL)
library(org.Hs.eg.db)
library(enrichplot)
library(biomaRt)
library(dplyr)
library(ggplot2)
```

```{r}
#load in the dataframes
GSK591 <-  read.table("C:\\Users\\maxim\\Dropbox (EinsteinMed)\\CloudStation\\PRMT\\RNAseq\\Dataframes\\completefile_Control_vs_GSK591_res.05_FC0_genename.txt", header = TRUE)
MS023 <-  read.table("C:\\Users\\maxim\\Dropbox (EinsteinMed)\\CloudStation\\PRMT\\RNAseq\\Dataframes\\completefile_Control_vs_MS023_res.05_FC0_genename.txt", header = TRUE)

Transcriptome <- list(GSK591 = GSK591, MS023 = MS023)

Transcriptome <- lapply(Transcriptome, function(x) {x <- x[c(9,3,7)]})
Transcriptome <- lapply(Transcriptome, function(x) x[!is.na(x$entrez),])
Transcriptome <- lapply(Transcriptome, function(x) x[!is.na(x$padj),])
Transcriptome <- lapply(Transcriptome, function(x) {x[x$padj == 0, 'padj'] <- 1e-300;x})
Transcriptome <- lapply(Transcriptome, function(x) {x <- x[x$padj < 0.05,];x})
Transcriptome <- lapply(Transcriptome, function(x) x[order(abs(x$padj)),])
Transcriptome <- lapply(Transcriptome, function(x) {x <- x[1:300,];x})
Transcriptome <- lapply(Transcriptome, function(x) {x$group <- "RNAseq";x})
#dataframes <- lapply(dataframes, function(x) {x$group[x$log2FoldChange > 0] <- "upregulated";x})
#dataframes <- lapply(dataframes, function(x) {x$group[x$log2FoldChange < 0] <- "downregulated";x})

df <- Transcriptome %>% bind_rows(.id = 'df')

```

```{r}
require(clusterProfiler)
formula_GO_BP <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'BP', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
formula_GO_MF <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'MF', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
formula_GO_CC <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'CC', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)

dotplot(formula_GO_BP, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

dotplot(formula_GO_MF, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

dotplot(formula_GO_CC, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
```

```{r}
#load in the dataframes
GluC <- read.csv("C:/Users/maxim/OneDrive - Montefiore Medicine/Shechter/DEP/Notebooks/GluC/SumPeptide/Input_Peptide_Tidy_DEP_Gluc_SumPeptides_WideFormat.csv")
Trypsin <-  read.csv("C:/Users/maxim/OneDrive - Montefiore Medicine/Shechter/DEP/Notebooks/Trypsin/SumPeptide/Input_Peptide_Tidy_DEP_Trypsin_SumPeptides_WideFormat.csv")
data <- rbind(GluC, Trypsin)
GSK591 <- data[data$GSK591_input_vs_DMSO_input_p.val < 0.05,]
MS023 <- data[data$MS023_input_vs_DMSO_input_p.val < 0.05,]

colnames(GSK591)[22] <- 'padj'
colnames(MS023)[27] <- 'padj'
colnames(GSK591)[20] <- 'log2FoldChange'
colnames(MS023)[25] <- 'log2FoldChange'

Proteome <- list(GSK591 = GSK591, MS023 = MS023)

mart <- useMart(biomart="ENSEMBL_MART_ENSEMBL", host="grch37.ensembl.org", path="/biomart/martservice", dataset="hsapiens_gene_ensembl")
Proteome_symbol <- lapply(Proteome, function(x) getBM(filters= "hgnc_symbol", attributes= c("hgnc_symbol","entrezgene_id"),values=x[2],mart= mart))
Proteome_symbol <- lapply(Proteome_symbol, setNames, c("name","entrez"))
Proteome <-mapply(merge, Proteome, Proteome_symbol, SIMPLIFY = FALSE)

Proteome <- lapply(Proteome, function(x) {x <- x[c(37, grep('log2FoldChange',colnames(x)),grep('padj',colnames(x)))];x})
Proteome <- lapply(Proteome, function(x) x[!duplicated(x$entrez),])
Proteome <- lapply(Proteome, function(x) x[order(abs(x$padj)),])
Proteome <- lapply(Proteome, function(x) x[!is.na(x$entrez),])
Proteome <- lapply(Proteome, function(x) {x <- x[1:300,];x})
Proteome <- lapply(Proteome, function(x) {x$group <- "Proteome";x})
#dataframes <- lapply(dataframes, function(x) {x$group[x$log2FoldChange > 0] <- "upregulated";x})
#dataframes <- lapply(dataframes, function(x) {x$group[x$log2FoldChange < 0] <- "downregulated";x})

df <- Proteome %>% bind_rows(.id = 'df')
```



```{r}
require(clusterProfiler)
formula_GO_BP <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'BP', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
formula_GO_MF <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'MF', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)
formula_GO_CC <- compareCluster(entrez~df, data = df, fun="enrichGO",  OrgDb = org.Hs.eg.db, ont = 'CC', pAdjustMethod = "BH", pvalueCutoff = 0.05, qvalueCutoff = 0.05)

dotplot(formula_GO_BP, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

dotplot(formula_GO_MF, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))

dotplot(formula_GO_CC, showCategory = 20)+ scale_color_gradient(low = '#E76F00', high='#55439B' ) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  theme(axis.text.x=element_text(angle=45, hjust=1))
```
